common/util.go

```diff
@@ -605,3 +605,72 @@ func NewCliHelper() CliHelper {
 		cZones:            make(map[string]string),
 	}
 }
+
+var overrideValueByPrefixLogMap = make(map[string]struct{})
+
+// OverrideValueByPrefix takes a list of override rules in the form 'prefix=val' and a given string, and determines the most specific rule
+// that applies to the given string. It then replaces the given default value with the override value. logFn is a logging closure that
+// allows lazy instatiation of a logger interface to log error conditions and override status. valName is used for logging purposes, to
+// differentiate multiple instantiations in the same context
+func OverrideValueByPrefix(logFn func() bark.Logger, path string, overrides []string, defaultVal int64, valName string) int64 {
```

Comment:
I looked at both of those places, and I don't like them for this. All the functions in those areas are receivers, and this doesn't seem like it should be one.

The function is pretty generic, it could be used to apply prefix rules for dconfig, yaml, CLI, etc.

I'd like to leave it here.